### DISCUSSION TITLE
Why is a JSON file generated when calling getStaticProps in Next.js?

### DISCUSSION THREAD

**Discussion by OGreeni** [vercel/next.js#40145] [Help] [answered]
I've been reading through the [Next.js docs][1] and came across this:

> Next.js generates a JSON file holding the result of running `getStaticProps`. The JSON file will be used in client-side routing through `next/link` or `next/router`. When you navigate to a page that's pre-rendered using `getStaticProps`, Next.js fetches this JSON file (pre-computed at build time) and uses it as the props for the page component.

My question is, why is this JSON file needed? If the page is rendered into static HTML at build time using the props provided by `getStaticProps`, why does the page component exist at runtime and needs access to those props?

[1]: https://nextjs.org/docs/basic-features/data-fetching/get-static-props#statically-generates-both-html-and-json

**Comment by icyJoseph** [comment, marked as answer]
For SPA navigation.

If you are in page `A`, and then navigate with SPA transitions to page `B`, given that `B` uses `getStaticProps`, then Next.js fetches the JS file that defines the structure of `B`, and the JSON data that is used as props to it. Notice that there's no HTML document change. Things stay the same. 

> Navigating with SPA transitions is the technique made popular in recent years where the browser doesn't really navigate away to the next page, but rather, only the URL changes, JavaScript takes care of changing what's being rendered.

However if you land on page `B`, the HTML for the page contains all the stuff it needs.

**Reply by OGreeni** [reply]
Got it. So only the initial page receives its rendered HTML file, and then the app behaves like a SPA?

**Reply by icyJoseph** [reply]
Roughly speaking yes. As long as you use the `next/link`, `next/router` pair of routing utilities.
